homeassistant/config_entries.py

```diff
@@ -2514,6 +2599,105 @@ def _async_get_config_entry(self, config_entry_id: str) -> ConfigEntry:
 
         return entry
 
+
+class ConfigSubentryFlowManager(
+    data_entry_flow.FlowManager[SubentryFlowResult], _ConfigSubFlowManager
+):
+    """Manage all the config subentry flows that are in progress."""
+
+    _flow_result = SubentryFlowResult
+
+    async def async_create_flow(
+        self,
+        handler_key: str,
+        *,
+        context: dict[str, Any] | None = None,
+        data: dict[str, Any] | None = None,
+    ) -> ConfigSubentryFlow:
+        """Create a subentry flow for a config entry.
+
+        Entry_id and flow.handler is the same thing to map entry with flow.
+        """
+        entry = self._async_get_config_entry(handler_key)
+        handler = await _async_get_flow_handler(self.hass, entry.domain, {})
+        return handler.async_get_subentry_flow(entry)
+
+    async def async_finish_flow(
+        self,
+        flow: data_entry_flow.FlowHandler[SubentryFlowResult],
+        result: SubentryFlowResult,
+    ) -> SubentryFlowResult:
+        """Finish a subentry flow and add a new subentry to the configuration entry.
+
+        Flow.handler and entry_id is the same thing to map flow with entry.
+        """
+        flow = cast(ConfigSubentryFlow, flow)
+
+        if result["type"] != data_entry_flow.FlowResultType.CREATE_ENTRY:
+            return result
+
+        entry = self.hass.config_entries.async_get_entry(flow.handler)
+        if entry is None:
+            raise UnknownEntry(flow.handler)
+
+        if "subentry_id" not in result or not isinstance(
+            subentry_id := result["subentry_id"], str
+        ):
+            raise HomeAssistantError("subentry_id must be a string")
+
+        if subentry_id in entry.subentries:
+            raise data_entry_flow.AbortFlow("already_configured")
+
+        self.hass.config_entries.async_update_entry(
+            entry,
+            subentries=entry.subentries
+            | {
+                subentry_id: ConfigSubentry(
+                    MappingProxyType(result["data"]), result["title"]
+                )
+            },
+        )
+
+        result["result"] = True
+        return result
```

Comment:
This method should trigger a set up the new subentry, using a new hook on the component. Hook should only be used when a subentry is created, not during setup of a config entry.